debian/rules

```diff
@@ -62,6 +67,14 @@ DEPENDS += ubuntu-dbgsym-keyring,
 #
 DEPENDS += delphix-extra,
 
+# Platform-specific dependencies
+DEPENDS.aws =
+DEPENDS.azure = walinuxagent,
```

Comment:
Do you need to make changes to the `postinst` file since it will try to disable a service that may or may not be installed or does this error out silently and just works?

Reply:
The `postinst` script does fail, but it turns out the failure is ignored by live-build. Rather than have this error in the build, I've just removed walinuxagent for now, since my understanding is that it is unused even on Azure. Once we get it working correctly on Azure we can add the package back.

Reply:
I see, so while we aren't using the daemon, we still want the package for the files it provides. I can add this back in.

Reply:
I think the correct answer is to remove the "disable" of the waagent from the script since it will only be installed on the VHD variant and it can be left in an enabled state. We still need to fix the cloudinit portion of it but it's not worse than the current state of things. What do you think?

Reply:
I've just recently learned about systemd's concept of "vendor presets". We could leverage this; we can set services to be disabled by default even if they don't exist (e.g. use the same preset value for all cloud platforms). See [here](https://www.freedesktop.org/software/systemd/man/systemd.preset.html) for more details.

Arguably we should do this for all services, and stop calling `systemctl` from the `postinst` script (I'm working on doing this for the virtualization package right now).

Reply:
Here's an example of what I mean (it's untested): https://github.com/delphix/delphix-platform/pull/34

Reply:
@grwilson I'm not sure what the consequences of leaving it enabled are, but if we can do that, that seems like the simplest option. If not, I can easily disable it using the method Prakash mentioned, or by just disabling it on Azure, making use of DPKG_MAINTSCRIPT_PACKAGE in the post install script to figure out what platform we are on.

Reply:
I think leaving it enabled should be fine since we still have to do some work to get the azure agent working correctly. Leaving it enabled won't make things worse.